imod/mf6/riv.py

```diff
@@ -144,19 +144,17 @@ def __init__(
         validate: bool = True,
         repeat_stress=None,
     ):
-        super().__init__(locals())
-        self.dataset["stage"] = stage
-        self.dataset["conductance"] = conductance
-        self.dataset["bottom_elevation"] = bottom_elevation
+        locals_at_init = locals()
```

Comment:
I assume that with this new functionality, many package constructors will contain the same code (a few lines to merge all the variables and to validate).
So they could be moved to PkgBase, but we need them in the package source file not for the code but for the argument list in the function header.  That is a bit counterintuitive. Another option would be to fill a dictionary in each package source file with the variable names and types its constructor needs. Then the constructor could go into pkgbase, and use these package-specific variables (a few packages would still have a custom contructor but that's fine) 
Or we could make a separate file with details of the datamodel, which lists for each package what goes into the package along with types, validation hints, and preferred regridding method. (In this approach we could also split what validation requirements and function arguments come from Modflow and what requirements and function arguments are imod-python specific). This would clean up the source files considerably, because the package metadata often is a considerable section of the package source file. 

Overall, I like the way constructors can be made shorter with this approach ( the one for the lake package is 60 lines, and with these changes, maybe 6?) but the downside is that it becomes (slightly) less transparent how things work for new developers.

Reply:
A similar approach could be:

```
def __init__(...):
  arg_dict = args2dict(locals())
    if concentration is None:
      del arg_dict["concentration"]
      del arg_dict["concentration_boundary_type"]

      super().__init__(arg_dict)
   ....
```

```
def args2dict(arguments):
    arguments.pop("self", None)
    arguments.pop("__class__", None)

    return arguments

```

A different approach could be to define a decorator to wrap the init method

```
@pkg_init
def __init__(...):
  ...
```


```
def pkg_init(init):
    @wraps(init)
    def merge_init(self, *args, **kwargs):
        args_dict = inspect.signature(self.__init__).bind_partial(*args, **kwargs).arguments
        for k, v in inspect.signature(self.__init__).parameters.items():
            if k not in args_dict and v.default is not inspect.Parameter.empty:
                args_dict[k] = v.default

        super(type(self), self).__init__(args_dict)
        return init(self, **args_dict)

    return merge_init
```

Reply:
Thanks for the wrapper idea, I implemented that.